sphinx/util/pycompat.py

```diff
@@ -103,3 +103,40 @@ def execfile_(filepath, _globals):
         else:
             raise
     exec_(code, _globals)
+
+# ------------------------------------------------------------------------------
+# Internal module backwards-compatibility
+
+import warnings
+
+from six import class_types
```

Comment:
I have one proposal. 
`class_types` should be also deprecated. It seems that there is no reason to exclude `class_types` from deprecation.

(I think it is not a big issue, it just a meaning of my first comment. sorry for less explanation.)